### PR TITLE
FH-4207 Add Keycloak walkthrough

### PR DESCRIPTION
**Pull request by aidenkeating**

### CODE REVIEW

**Comment by aidenkeating** [comment]
@matzew @philbrookes Would you mind taking a look?

File: docs/walkthroughs/keycloak.adoc [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+Mobile Services.
+
+To integrate Keycloak with the FeedHenry Sync Server navigate to the MCP and
+then to the Mobile Enabled Service that will be integrated with Keycloak.
+In the Service Integrations tab click Create Integration for Keycloak.
+
+Creating an integration will mount the secret named after a service, in this
+case `keycloak`, into the Mobile Enabled Service. The contents of the secret
+will then be available in the file system of the service. The files that are
+created will be in the directory `/etc/secrets/{service-name}`. For Keycloak
```
Comment (matzew) [inline docs/walkthroughs/keycloak.adoc]: directory, where ? not sure that is clear

File: docs/walkthroughs/keycloak.adoc [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+When, for example, the FeedHenry Sync service integrates with Keycloak it will
+restart and detect that keycloak has been integrated by looking for whether the
+mounted files exist. If they do then the service will protect itself using
+Keycloak.
+
+Integrate Keycloak with a Mobile App
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once the Keycloak service is successfully provisioned it will be available to
+mobile apps using the Mobile Core Library. This library can be used to obtain
```
Comment (matzew) [inline docs/walkthroughs/keycloak.adoc]: > Mobile Core Library

let's add a link?

**Comment by matzew** [comment]
As discussed on IRC, for me the following sentence is a bit unclear:

> "Once Keycloak has provisioned it can be integrated with another Mobile Service e.g. FeedHenry Sync Server. The Keycloak service is not available to all other Mobile Services. 

@aidenkeating and I did discuss a better version on our IRC channel

**Comment by aidenkeating** [comment]
@matzew Would you mind taking another look?

**Pull request merged**